loopy/kernel/creation.py

```diff
@@ -2563,6 +2556,19 @@ def make_function(domains, instructions, kernel_data=None, **kwargs):
     knl = add_inferred_inames(knl)
     from loopy.transform.parameter import fix_parameters
     knl = fix_parameters(knl, **fixed_parameters)
+
+    # -------------------------------------------------------------------------
+    # Ordering dependency:
+    # -------------------------------------------------------------------------
+    # Must duplicate inames after adding all the inames.
```

Comment:
Could you explain a bit more (in the comment)? The nature of the dependency isn't obvious to me from the comment.

Reply:
LGTM!